### PR TITLE
Part/Toponaming: makeElementWires - import from LS3 fork

### LINKED ISSUE #14436: PartDesign: No planar face in AttachEngine3D::calculateAttachedPlacement
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

No planar face in AttachEngine3D::calculateAttachedPlacement()

In console i get this:

```
<PropertyLinks> PropertyLinks.cpp(453): Unnamed#Sketch001.Support missing element reference Unnamed#Pad ;g1841;SKT;:H1160,E;FAC;:H1160:4,F;:G0;XTR;:H1160:8,F.Face6
Sketch001: No planar face in AttachEngine3D::calculateAttachedPlacement()!
``` 

### Full version info

```shell
OS: Arch Linux (XFCE/xfce)
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.37573 (Git)
Build type: Release
Branch: main
Hash: 91aac05021023c946dc1545f2b13512ccf326550
Python 3.12.3, Qt 5.15.14, Coin 4.0.2, Vtk 9.3.0, OCC 7.7.2
Locale: English/United States (en_US)
```

### Subproject(s) affected?

None

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by CalligaroV**
Change the definitions of the `makeElementWires` methods, aligning them with LS3 fork.

For reference:
https://github.com/realthunder/FreeCAD/blob/0e9cf4aa7c7bac94a83edd07b4bd74a71a26446c/src/Mod/Part/App/TopoShapeEx.cpp#L3004
https://github.com/realthunder/FreeCAD/blob/0e9cf4aa7c7bac94a83edd07b4bd74a71a26446c/src/Mod/Part/App/TopoShapeEx.cpp#L3181

In my tests this fixes #14436

### CODE REVIEW

**Comment by Syres916** [comment]
Thanks @CalligaroV, I've got some other tests to go through and hopefully this PR fixes a few of those as well.

**Comment by CalligaroV** [comment]
Thank you too @Syres916 for looking at this!

I analyzed the test you proposed but it fails also with the modifications of this PR.
It looks like a combination of #14129 and https://forum.freecad.org/viewtopic.php?t=87966.

The first issue looks not solved also in RealThunder's LinkStage fork (https://github.com/FreeCAD/FreeCAD/issues/14129#issuecomment-2121142494) while the second needs some modifications in the main branch (I tried with LinkStage and there it looks working) but I haven't figured out yet how to solve it.

Therefor I would say that for sure the test you added should be included but not here, in future PRs.

If you want you can open a new issue for https://forum.freecad.org/viewtopic.php?t=87966, I haven't seen an issue here on GH reporting this specific case (but maybe I simply missed it).

Thanks again!

**Comment by chennes** [comment]
@bgbsww can you give this a quick look?

**Review by bgbsww — APPROVED** [review]
Looks good!

File: src/Mod/Part/App/TopoShapeExpansion.cpp [hunk 1]
```diff
@@ -3083,17 +3079,19 @@ TopoShape& TopoShape::makeElementWires(const TopoShape& shape,
         std::vector<TopoShape> wires;
         for (int i = 1; i <= hWires->Length(); i++) {
             auto wire = hWires->Value(i);
-            wires.emplace_back(Tag, Hasher, wire);
+            wires.push_back(TopoShape(Tag,Hasher,wire));
```
Comment (chennes) [inline src/Mod/Part/App/TopoShapeExpansion.cpp:3082]: Why the change from `emplace_back` here?

**Comment by bgbsww** [comment]
@CalligaroV are you around to update this?  Looks at a glance like some .Support needs to be .AttachmentSupport in tests (I assume someone removed the deprecated property).

File: src/Mod/Part/App/TopoShapeExpansion.cpp (hunk 1 above)
Comment (chennes) [inline src/Mod/Part/App/TopoShapeExpansion.cpp:3082]:
```suggestion
            wires.emplace_back(Tag,Hasher,wire);
```

File: src/Mod/PartDesign/PartDesignTests/TestTopologicalNamingProblem.py [hunk 2]
```diff
@@ -1469,6 +1469,433 @@ def create_t_sketch(self):
...
+        self.Doc.recompute()
+
+        self.Pad001.Type = "Length"
+        self.Pad001.Length = "4 mm"
+        self.Doc.recompute()
+        """Assumption: the warning <PropertyLinks> PropertyLinks.cpp(453):
+         PartDesignTestTNP1#Sketch002.AttachmentSupport missing element
+         reference PartDesignTestTNP1#Pad001 ;g815v1;SKT;:H976,V;:L#2;PSM;
+        :H976:9,E;:L#8;PSM;:H976:9,F;:H-977,F.Face11 is only temporary and can be ignored."""
+        self.assertTrue(self.Sketch002.Support[0][1][0] == "Face11")
```
Comment (chennes) [inline src/Mod/PartDesign/PartDesignTests/TestTopologicalNamingProblem.py:1896]:
```suggestion
        self.assertTrue(self.Sketch002.AttachmentSupport[0][1][0] == "Face11")
```
